packages/@aws-cdk/aws-gamelift/lib/standalone-matchmaking-configuration.ts

```diff
@@ -0,0 +1,119 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import * as gamelift from './gamelift.generated';
+import { MatchmakingConfigurationProps, MatchmakingConfigurationBase, IMatchmakingConfiguration } from './matchmaking-configuration';
+
+/**
+ * Properties for a new standalone matchmaking configuration
+ */
+export interface StandaloneMatchmakingConfigurationProps extends MatchmakingConfigurationProps {}
+
+/**
+ * A FlexMatch matchmaker process does the work of building a game match.
+ * It manages the pool of matchmaking requests received, forms teams for a match, processes and selects players to find the best possible player groups, and initiates the process of placing and starting a game session for the match.
+ * This topic describes the key aspects of a matchmaker and how to configure one customized for your game.
+ *
+ * @see https://docs.aws.amazon.com/gamelift/latest/flexmatchguide/match-configuration.html
+ *
+ * @resource AWS::GameLift::MatchmakingConfiguration
+ */
+export class StandaloneMatchmakingConfiguration extends MatchmakingConfigurationBase {
+
+  /**
+     * Import an existing matchmaking configuration from its name.
+     */
+  static fromStandaloneMatchmakingConfigurationName(scope: Construct, id: string, matchmakingConfigurationName: string): IMatchmakingConfiguration {
+    return this.fromMatchmakingConfigurationAttributes(scope, id, { matchmakingConfigurationName });
+  }
+
+  /**
+     * Import an existing matchmaking configuration from its ARN.
+     */
+  static fromStandaloneMatchmakingConfigurationArn(scope: Construct, id: string, matchmakingConfigurationArn: string): IMatchmakingConfiguration {
+    return this.fromMatchmakingConfigurationAttributes(scope, id, { matchmakingConfigurationArn });
+  }
+
+  /**
+   * The Identifier of the matchmaking configuration.
+   */
+  public readonly matchmakingConfigurationName: string;
+  /**
+   * The ARN of the matchmaking configuration.
+   */
+  public readonly matchmakingConfigurationArn: string;
+  /**
+   * The notification target for matchmaking events
+   */
+  public readonly notificationTarget?: sns.ITopic;
+
+  constructor(scope: Construct, id: string, props: StandaloneMatchmakingConfigurationProps) {
+    super(scope, id, {
+      physicalName: props.matchmakingConfigurationName,
+    });
+
+    if (props.matchmakingConfigurationName && !cdk.Token.isUnresolved(props.matchmakingConfigurationName)) {
+      if (props.matchmakingConfigurationName.length > 128) {
+        throw new Error(`Matchmaking configuration name can not be longer than 128 characters but has ${props.matchmakingConfigurationName.length} characters.`);
+      }
+
+      if (!/^[a-zA-Z0-9-\.]+$/.test(props.matchmakingConfigurationName)) {
```

Comment:
As with my last comment I'm unsure if this is the best way to handle these sorts of checks but it works. I will get a second opinion though

Reply:
I'm OK with this as is, but that's only because I don't see the value that a utility would add here. Having to say `cdk.regex.of(//).test(...)` is just more words to do exactly what `//.test(...)` does.